src/hardhat-deploy-ethers.ts

```diff
@@ -0,0 +1,94 @@
+/* Imports: External */
+import { Contract } from 'ethers'
+import { Provider } from '@ethersproject/abstract-provider'
+import { Signer } from '@ethersproject/abstract-signer'
+import { HardhatRuntimeEnvironment } from 'hardhat/types'
+
+export const deployAndRegister = async ({
+  hre,
+  name,
+  args,
+  contract,
+}: {
+  hre: HardhatRuntimeEnvironment
+  name: string
+  args: any[]
+  contract?: string
+}) => {
+  const { deploy } = hre.deployments
+
+  // TODO: Cache these 2 across calls?
+  const { deployer } = await hre.getNamedAccounts()
+  const Lib_AddressManager = await getDeployedContract(
+    hre,
+    'Lib_AddressManager',
+    {
+      signerOrProvider: deployer,
+    }
+  )
```

Comment:
This is called twice in each script now, once in the external deploy script and once here.